### DISCUSSION TITLE
[HELP] ¿Cómo funciona el filter de multiple choice de un survey?

### DISCUSSION THREAD

**Discussion by ValeMP** [Cotalker/documentation#50] [2. Questions and Answers] [answered]
Hola!

Quería saber si me podrían explicar un poco sobre el filtro de un multiple choice en un survey. Me gustaría saber si en la casilla de al lado, se configura la operación o si siempre sería "equal".

Les adjunto una imagen! 
![Screen Shot 2021-04-08 at 13 57 54](https://user-images.githubusercontent.com/42095617/114075032-11123900-9873-11eb-8a35-a93b8737aab4.png)

**Comment by acedward** [comment, marked as answer]
#56 Fue creado para acutalizar la documentación de este punto

**Reply by acedward** [reply]
1. subproperty-channel
**filter data**  va el __code__ de la propertyType a buscar en el canal, las opciones serán sus subproperties.

2. subproperty-user
**filter data**  va el __code__ de la propertyType a buscar en el user, las opciones serán sus subproperties.

3. subproperty
(pending - estoy investigando) 

E.g., 
Configuración Collection **marcas**
Filter **subproperty-channel**
Filter-data **categoria_gondola**

Entonces solamente se muestran, en la lista propiedades, las cuales 
1. son del tipo **marcas** 
2. y al mismo tiempo sean subpropiedades de cualquier propiedad del canal (en el cual se levanta el formulario) que sea de tipo **categoría_gondola**

Esto es necesario, porque el canal puede tener muchas propiedades asignadas, pero necesitamos saber cual es el tipo que debemos usar para filtrar, esto es el **filter-data**

**Reply by acedward** [reply]
@guillevalin no entendí como funciona el **subproperty** solo la configucación de los selectores con este tipo de atributo. Nos puedes ayudar?

**Reply by NicolasPLG** [reply]
adjunto un ss visual de un caso con subproperty-channel
![image](https://user-images.githubusercontent.com/61419471/114229321-aa118480-9945-11eb-87ec-208fff98962b.png)

Con respecto al **subproperty**, pensé que servía para indicar que elemento contiene una property y me visualizara esos elementos: Ejemplo:

Tengo las properties con sus sub properties:
PA / SPA, SPB, SPC
PB / SPA, SPB
PC / SPC

Si colocara el filtro subproperties y la condición "SPC". El selector me debería mostrar PA y PC, pero creo que no es eso porque lo probé y no me funcionó :(.

Espero que sea un aporte a la sección

**Reply by guillevalin** [reply]
@acedward el modo **subproperty** permite escoger subpropiedades de otro campo de tipo propiedad en el mismo formulario.

A modo de ejemplo tengo este selector de propiedades:
![image](https://user-images.githubusercontent.com/34345309/117481122-80da1900-af30-11eb-87d8-fc5a7d516373.png)

Y quisiera implementar otro selector que muestre las subpropiedades de la selección anterior:
![image](https://user-images.githubusercontent.com/34345309/117479992-ffce5200-af2e-11eb-848b-4863996a508a.png)

Para ello se configura con el identifier de la pregunta que tiene la propiedad padre así como está en el ejemplo.

El resultado será este:
![image](https://user-images.githubusercontent.com/34345309/117480045-14124f00-af2f-11eb-8e94-1a3831e55c79.png)

Saludos!

**Reply by acedward** [reply]
Tree remplazaría esto completamente?

**Reply by acedward** [reply]
Se ha creado el issue : https://github.com/Cotalker/documentation/issues/156 
Para documentar esto

**Reply by guillevalin** [reply]
> Tree remplazaría esto completamente?

Quizás hay casos donde prefieres que funcionen de manera independiente los selectores y no necesariamente como una selección de arbol, pero efectivamente tree es una solución mucho más amigable para implementar selecciones de elementos dependientes entre ellos.
